### PR TITLE
Add support for the payment_intent resource

### PR DESCRIPTION
**Pull request by remi-stripe**
This resource is gated so for now tests will be stubbed.

cc @michelle-stripe @jenan-stripe @stripe/api-libraries

### CODE REVIEW

File: src/main/java/com/stripe/model/PaymentIntentCollection.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.stripe.model;
+
+public class PaymentIntentCollection extends StripeCollection<Payout> {
```
[the remark this answers is not in the document]
Comment (remi-stripe) [reply src/main/java/com/stripe/model/PaymentIntentCollection.java]: yes it's just a WIP branch to start discussion on naming

File: src/main/java/com/stripe/model/PaymentIntent.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  String confirmationMethod;
+  Long created;
+  String currency;
+  @Getter(AccessLevel.NONE) @Setter(AccessLevel.NONE) ExpandableField<Customer> customer;
+  Boolean livemode;
+  @Getter(onMethod = @__({@Override})) Map<String, String> metadata;
+  PaymentIntentSourceAction nextSourceAction;
+  @Getter(AccessLevel.NONE) @Setter(AccessLevel.NONE) ExpandableField<Account> onBehalfOf;
+  String receiptEmail;
+  String returnUrl;
```
Comment (remi-stripe) [inline src/main/java/com/stripe/model/PaymentIntent.java]: Can you confirm it's the right way to write url and that it does not need anything custom?

File: src/main/java/com/stripe/model/PaymentIntentSourceActionDeserializer.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  @Override
+  public PaymentIntentSourceAction deserialize(JsonElement json, Type typeOfT,
+      JsonDeserializationContext context)
+      throws JsonParseException {
+    PaymentIntentSourceAction sourceAction = new PaymentIntentSourceAction();
+    JsonObject jsonObject = json.getAsJsonObject();
+    String type = jsonObject.get("type").getAsString();
+    sourceAction.setType(type);
+    Class<? extends PaymentIntentSourceActionValue> cl = objectMap.get(type);
+    if (cl != null) {
```
Comment (remi-stripe) [inline src/main/java/com/stripe/model/PaymentIntentSourceActionDeserializer.java]: added this and confirmed that if I add a wrong `type` the deserializer does not crash.

File: src/main/java/com/stripe/model/ShippingDetails.java [hunk 4]
```diff
@@ -9,6 +9,8 @@
 @EqualsAndHashCode(callSuper = false)
 public final class ShippingDetails extends StripeObject {
   protected Address address;
+  protected String carrier;
   protected String name;
   protected String phone;
+  protected String trackingNumber;
```
Comment (remi-stripe) [inline src/main/java/com/stripe/model/ShippingDetails.java]: This is a bit unrelated but we did not support those properties even though they are valid on `Order`. They are not valid on `Charge`. We likely should revisit `shipping` as a whole in the API though.

**Comment by remi-stripe** [comment]
@ob-stripe Thank you for the careful review! Damn I missed a few things! Can you have a look?

**Comment by ob-stripe** [comment]
Released in 5.46.0.

**Pull request merged**